### PR TITLE
Add python intro and basics

### PR DESCRIPTION
**Pull request by alayek**
This is the starting of wiki entries for Python. Python challenges would be based on this.

### CODE REVIEW

**Comment by Rafase282** [comment]
I took a quick look and the first thign is to change the filename to have the first character uppercase in all words.

I will take a closes look at the content itself soon in a few minute. @alayek

File: Python-Basics.md [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+
+*Objects* represent a logical grouping of attributes. Attributes are data and/or functions. When an object is created in Python it is created with an *identity*, *type*, and *value*.
+
+In other languages, *primitives* are *values* that have no *properties* (attributes). For example, in javascript `undefined`, `null`, `boolean`, `string`, `number`, and `symbol` (new in ECMAScript 2015) are primitives.
+
+In Python, there are no primitives. `None`, *booleans*, *strings*, *numbers*, and even *functions* are all *objects* regardless how they are created.
+
+We can demonstrate this using some built-in functions:
+
+* `id`:  https://docs.python.org/3/library/functions.html#id
```
Comment (Rafase282) [inline Python-Basics.md]: [`id`](https://docs.python.org/3/library/functions.html#id)

You could use that format for these. I think is always better to hide the links like this and looks better.

File: Python-Basics.md [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+>>> print(some_variable)
+1
+>>> some_variable = "Hello campers!"
+>>> print(some_variable)
+Hello campers!
+```
+
+## Keywords are Reserved
+
+Python has a list of keywords that cannot be used as identifiers (variable names):
```
Comment (Rafase282) [inline Python-Basics.md]: Python has a list of [keywords](https://docs.python.org/3/reference/lexical_analysis.html#keywords) that cannot be used as identifiers (variable names).

File: Python-Basics.md [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+    """Triple quoted strings can span multiple lines.
+    Unescaped "double" and 'single' quotes in triple quoted strings are retained."""
+```
+
+```python
+    '''Triple quoted strings can be 'single'or "double" quotes.
+    Unescaped newlines are also retained.'''
+```
+#### Reference:
+https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str
```
Comment (Rafase282) [inline Python-Basics.md]: [Text Sequence Type *str*](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str)

File: Python-Introduction.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+# Introduction
+
+## What is Python?
+
+https://www.python.org/doc/essays/blurb/
```
Comment (Rafase282) [inline Python-Introduction.md]: Please use markdown links as explained earlier.

**Comment by Rafase282** [comment]
@alayek Everything looks good, just the few points I made, about the links and the already suggested changes.

**Comment by alayek** [comment]
Done @Rafase282 

Please check

File: Python-Basics.md [hunk 5]
```diff
@@ -0,0 +1,711 @@
...
+>>> a(a)
+['__call__', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__name__', '__ne__', '__new__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__self__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__text_signature__']
+```
+
+Resources:
+- https://www.jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/
+
+## Variables, Names, and Binding
+
+Having *objects* isn't useful unless there is a way to use them. In order to use an *object*, there must be a way to reference them. In Python this is done by **binding** objects to **names**. A detailed overview of can be found here: https://docs.python.org/3/reference/executionmodel.html
```
Comment (Rafase282) [inline Python-Basics.md]: Please use markdown link here.

File: Python-Basics.md [hunk 6]
```diff
@@ -0,0 +1,711 @@
...
+
+One way this is done is by using an *assignment statement*, https://docs.python.org/3/reference/simple_stmts.html#assignment-statements. This is commonly called *assigning a variable* in the context of Python. If speaking about programming in the context of other languages, **binding** an *object* to a **name** may be more precise.
```
Comment (Rafase282) [inline Python-Basics.md]: Please use markdown link here.

File: Python-Basics.md [hunk 7]
```diff
@@ -0,0 +1,711 @@
...
+>>> break = "Hello campers!"
+File "<stdin>", line 1
+    break = "Hello campers!"
+            ^
+    SyntaxError: invalid syntax
+```
+
+## Code Blocks and Indentation
+
+https://docs.python.org/3/reference/lexical_analysis.html#indentation
```
Comment (Rafase282) [inline Python-Basics.md]: Seems to be a section not yet completed  but please use markdown link here.

File: Python-Basics.md [hunk 8]
```diff
@@ -0,0 +1,711 @@
...
+
+    def method_a(self):
+        """Single line summary of method_a."""
+        pass
+```
+String literals that start and end with `"""` that are not docstrings (not the first statement), can be used for multiline strings. They will not become `__doc__` attributes. If they are not assigned to a variable, they will not generate bytecode. There is some discussion about using them as multiline comments found [here](http://stackoverflow.com/questions/7696924/multiline-comments-in-python).
+
+## Built-in Constants
+
+https://docs.python.org/3/library/constants.html
```
Comment (Rafase282) [inline Python-Basics.md]: Please use markdown link here.

File: Python-Basics.md [hunk 9]
```diff
@@ -0,0 +1,711 @@
...
+
+## Integers
+
+The theoretical domain for integers in python is negative infinity to infinity. In practice, integer values are limited by the amount of available memory.
+
+In Python 2, there was a distinction between **`int`**, numbers that fit in a 32 or 64 bit *C long*, and **`long`**,  numbers limited by available memory. Python 3 unified the two types into just **`int`**, more info in [PEP 237](https://www.python.org/dev/peps/pep-0237/).
+
+**`int` creation using integer literals**
+
+https://docs.python.org/3/reference/lexical_analysis.html#integer-literals
```
Comment (Rafase282) [inline Python-Basics.md]: Please use markdown link here.

File: Python-Introduction.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+* logging module, *logging*: https://docs.python.org/3/library/logging.html
+* debugging module, *pdb*: https://docs.python.org/3/library/pdb.html
+
+Just note that these exist for now.
+
+## Hello World!
+
+Going back to the docs, we can read about the `print` function, a [*built-in function*](https://docs.python.org/3/library/functions.html) of the [Python Standard Library](https://docs.python.org/3/library/index.html).
+
+https://docs.python.org/3/library/functions.html#print
```
Comment (Rafase282) [inline Python-Introduction.md]: Please use markdown link here.

**Pull request merged**